geoseong/hackatalk-mobile
# Search users

## Description
- Add query to collect users
- Attach apollo to collect actual data from server('hackatalk-server')
> I made my branch named **'feat/usermgmt-users'** instead of **'feat/usermgmt/users'** due to following error. please understand I think because I have already **'feat/usermgmt'** branch so cannot create sub-branch. it is just my case. maybe other contributor won't happen.
```
fatal: cannot lock ref 'refs/heads/feat/usermgmt/users': 'refs/heads/feat/usermgmt' exists; cannot create 'refs/heads/feat/usermgmt/users'
```

## Related Issues
[#106](https://github.com/dooboolab/hackatalk-mobile/issues/106)

## Tests
- remove mocking users in actual `SearchUser.ts` and add mocking users in test code
- ~try to make 100% on SearchUser~
- I ran following test script on local
  ```
  yarn test --coverage --watch src/components/screen/__tests__/SearchUser.test.tsx -u
  ```
- Add ["MockedProvider"](https://www.apollographql.com/docs/react/development-testing/testing/#mockedprovider) in test code
- make following warning message disappear using 
  - [`await act(() => wait());`](https://github.com/geoseong/hackatalk-mobile/blob/8e455f1f37d1224c3969f185d695914792bc2953/src/components/screen/__tests__/SearchUser.test.tsx#L313)
  ```
  console.error node_modules/react-native/Libraries/YellowBox/YellowBox.js:63
  Warning: An update to Screen inside a test was not wrapped in act(...).

  When testing, code that causes React state updates should be wrapped into act(...):

  act(() => {
    /* fire events that update state */
  });
  /* assert on the output */
  ```
### coverage
![스크린샷 2020-02-26 오전 12 13 31](https://user-images.githubusercontent.com/19166187/75262156-74880580-582f-11ea-8869-c8230f043dc7.png)

### TODO
- I've tried to cover [TestComponent](https://github.com/geoseong/hackatalk-mobile/blob/8e455f1f37d1224c3969f185d695914792bc2953/src/components/screen/__tests__/SearchUser.test.tsx#L252) with [MockedProvider](https://www.apollographql.com/docs/react/development-testing/testing/#mockedprovider) but didn't figure out how to cover

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk-mobile/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn lint && yarn tsc`
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
